writer/writer/postgresql_backend/sql_occ_locker_backend_service.py

```diff
@@ -95,27 +104,53 @@ def assert_fqfield_positions(self, fqfields: Dict[str, int]) -> None:
         self.raise_model_locked_if_match(query, query_arguments)
 
     def assert_collectionfield_positions(
-        self, collectionfields: Dict[str, int]
+        self, collectionfields: Dict[str, CollectionFieldLock]
     ) -> None:
         if not collectionfields:
             return
 
         query_arguments: List[Any] = []
         filter_parts = []
-        for collectionfield, position in collectionfields.items():
-            query_arguments.extend(
-                (
-                    collectionfield,
-                    position,
+        for collectionfield, cf_lock in collectionfields.items():
+            if isinstance(cf_lock, int):
+                query_arguments.extend(
+                    (
+                        collectionfield,
+                        cf_lock,
+                    )
                 )
-            )
-            filter_parts.append("(collectionfield=%s and position>%s)")
-        query = (
-            "select collectionfield from collectionfields where "
-            + " or ".join(filter_parts)
-            + " limit 1"
-        )
+                filter_parts.append("(cf.collectionfield=%s and e.position>%s)")
+            else:
+                collection = collection_from_collectionfield(collectionfield)
+
+                query_arguments.extend(
+                    (
+                        collection.replace("_", r"\_") + "%",
```

Comment:
Is the `%` right there? shoule locing `motion` also lock `motion_category`?

Reply:
You're right, there is a `.` missing. Will fix

Reply:
maybe write a small test, that the collection match is exact ;)

Reply:
Actually that check was redundant since the collection is also part of the collection field which is check in the same condition. Anyway, added a test for it